packages/core/src/pubsub/task-queue.ts

```diff
@@ -30,17 +30,23 @@ export interface TaskQueueLike {
  * PQueue with synchronous `add` and a common error-handler.
  */
 export class TaskQueue implements TaskQueueLike {
```

Comment:
nitpick: can we move this out of the `pubsub` directory?  Perhaps to a new `utils` directory?